File: backend/Food.fs
module Foods

[<CLIMutable>]
type Food = {
    Id : string
    Title : string
    Done : bool
    Date : int64
}

type FoodSave = Food -> Food

type FoodCriteria =
  | All

type FoodFind = FoodCriteria -> Food[]

type FoodDelete = string -> bool

File: backend/src/food.ts
import mongoose, { Schema, Document } from 'mongoose'

export interface IFood extends Document {
  id: string
  title: string
  done: boolean
  date: number
}

const FoodSchema: Schema = new Schema({
  title: { type: String, required: true },
  done: Boolean,
  date: Number
})

export const Food = mongoose.model<IFood>('Food', FoodSchema);